### PR TITLE
#544 RqChunkTest.java:40-43: Implement reading chunked message body

### PR DESCRIPTION
**Pull request by xupyprmv**
#544. I created `ChunkedInputStream` that provides `InputStream` interface to read  messages encoded with [chunked transfer coding](https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.6.1).

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Thanks for your pull request, let me find someone who can review it

**Comment by davvd** [comment]
@dmzaytsev review this one,please

**Comment by dmzaytsev** [comment]
@xupyprmv please put to the pull request description link to the original ticket

**Comment by xupyprmv** [comment]
@dmzaytsev PR description was updated.

**Comment by dmzaytsev** [comment]
@xupyprmv looks good to me
however I would suggest to avoid to make big PRs in the future
I guess you have spent more than 30 mins

**Comment by dmzaytsev** [comment]
@rultor merge

File: src/main/java/org/takes/rq/ChunkedInputStream.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    /**
+     * Read the CRLF terminator.
+     * @throws IOException If an IO error occurs.
+     */
+    private void readCRLF() throws IOException {
+        final int crsymbol = this.origin.read();
+        final int lfsymbol = this.origin.read();
+        if (crsymbol != '\r' || lfsymbol != '\n') {
+            throw new IOException(
+                "CRLF expected at end of chunk: " + crsymbol + "/" + lfsymbol
```
Comment (yegor256) [inline src/main/java/org/takes/rq/ChunkedInputStream.java]: @xupyprmv string concatenation is a bad idea, see http://www.yegor256.com/2014/06/19/avoid-string-concatenation.html

**Comment by yegor256** [comment]
@xupyprmv it's a new class and you're suppressing so many warnings... don't you think that it's better to make it clean in the first place? why committing such a dirty code to master?

**Comment by xupyprmv** [comment]
@yegor256 I removed all `@SuppressWarnings` except of `PMD.MissingBreakInSwitch`. I think it is a PMD bug, because code doesn't contain missing breaks.

Please, continue with CR.

**Comment by dmzaytsev** [comment]
@xupyprmv I would suggest submit a bug to qulice project and create a puzzle, that contains a link to the qulice ticket, to remove the suppression

> except of PMD.MissingBreakInSwitch. I think it is a PMD bug, because code doesn't contain missing breaks.

**Comment by xupyprmv** [comment]
@dmzaytsev thanks for advise. I refactored code to avoid nested switch and added break to default case. So, we don't have to add puzzle here.

**Comment by xupyprmv** [comment]
@yegor256 Please, continue with CR.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by xupyprmv** [comment]
@yegor256 please merge again

**Comment by xupyprmv** [comment]
@rultor merge

**Comment by xupyprmv** [comment]
@yegor256 Please, merge this PR.

**Comment by xupyprmv** [comment]
@yegor256 please merge

**Comment by xupyprmv** [comment]
@yegor256 please continue with merge

**Comment by xupyprmv** [comment]
@yegor256 initiate merge again, please

**Comment by yegor256** [comment]
@rultor merge

**Comment by davvd** [comment]
@elenavolokhova please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by dmzaytsev** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@dmzaytsev just added 10 mins to @elenavolokhova (for QA), payment ID is `78672172`

Many thanks! **20 mins** were added to your account in Transaction ID `78672185` (task took 588 hours and 17 mins)

review comments (c=5) added as a bonus

+20 added to your rating, at the moment it is: [+3629](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**